GoSmartSearch.go

```diff
@@ -31,7 +30,7 @@ func SearchInMaps(elements []map[string]string, term string, key string, toleran
 		return nil, err
 	}
 
-	var result []map[string]string
+	result := make([]map[string]string, len(sortedKeyValues))
```

Comment:
Whoops, that needs to be `result := make([]map[string]string, 0, len(sortedKeyValues))`